### PR TITLE
Client-side work for the Precontribution Reminder with first round test

### PR DESCRIPTION
**Pull request by jlieb10**
## What does this change?
This is a pre-contribution reminder feature that will be shown on Contributions epics on theguardian.com. This feature allows users of theguardian.com to submit their email address in order to receive a service email in the future (via Braze) that will remind them to contribute. In the first round of the test, we will be testing whether or not this affects the AV of the epic

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [X] Yes (please indicate your plans for DCR Implementation)

Plan is that this will be absorbed into the rest of the work to migrate the Epic to DCR
## Screenshots
epic

## What is the value of this and can you measure success?
Yes, we will be measuring whether or not this harms conversion rate

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [X] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [X] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [X] Locally
- [X] On CODE (optional)

### CODE REVIEW

File: static/src/stylesheets/module/reader-revenue/_epic-reminder.scss [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    .epic-reminder__reveal-reminder:checked ~ .epic-reminder__prompt-label {
+        display: none;
+    }
+
+    .epic-reminder__signup {
+        @include multiline(4, $brightness-86, top);
+        display: none;
+        line-height: initial;
```
Comment (tjsilver) [inline static/src/stylesheets/module/reader-revenue/_epic-reminder.scss]: Not sure if relevant here, but on support-frontend we get messages that using line-height is discouraged.

File: static/src/stylesheets/module/reader-revenue/_epic-reminder.scss [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            flex: 1 1 auto;
+        }
+    }
+
+    .epic-reminder__email-help-text {
+        @include fs-textSans(5);
+        display: block;
+        color: $news-main;
+        font-size: 17px;
+        line-height: 21px;
```
Comment (tjsilver) [inline static/src/stylesheets/module/reader-revenue/_epic-reminder.scss]: see above re line-height

**Review by tjsilver — APPROVED** [review]
Great work!

File: static/src/stylesheets/module/reader-revenue/_epic-reminder.scss (hunk 1 above)
Comment (jlieb10) [reply static/src/stylesheets/module/reader-revenue/_epic-reminder.scss]: so line-height is set by using the multiline mixin and i am overriding it here with 'initial' (which is essentially saying 'reset to default) for the reason why setting it is discouraged: it limits line height that can be set on children elements.

File: static/src/stylesheets/module/reader-revenue/_epic-reminder.scss (hunk 2 above)
Comment (jlieb10) [reply static/src/stylesheets/module/reader-revenue/_epic-reminder.scss]: this element doesn't have any children, so it's okay to use line height here (though obviously getting it from a typography mixin would be desirable, but we don't have this treatment anywhere else)

File: static/src/stylesheets/module/reader-revenue/_epic-reminder.scss (hunk 1 above)
Comment (tjsilver) [reply static/src/stylesheets/module/reader-revenue/_epic-reminder.scss]: 👍

File: static/src/javascripts/projects/common/modules/commercial/epic-reminder-email-signup.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    };
+
+    const sendReminderEvent = (): Promise<Response> => {
+        const isProd = config.get('page.isProd');
+
+        const email = fields.emailInput.value || '';
+
+        if (!isValidEmail(email)) {
+            setState('invalid');
+            return Promise.reject(new Error('Email is invalid'));
```
Comment (tomrf1) [inline static/src/javascripts/projects/common/modules/commercial/epic-reminder-email-signup.js]: what happens to this error?

**Review by tomrf1 — APPROVED** [review]
brilliant!

**Pull request merged**